src/js/pages/jobs/JobsTable.js

```diff
@@ -96,7 +100,22 @@ class JobsTable extends React.Component {
     });
   }
 
-  sortJobNames(prop, direction) {
+  getCompareFunctionByProp(prop) {
+    switch (prop) {
+      case "name":
+        return (a, b) => a.name.localeCompare(b.name);
+      case "status":
+        return (a, b) =>
+          JobStates[a.status].sortOrder - JobStates[b.status].sortOrder;
+      case "lastRun":
+        return (a, b) =>
+          JobStatus[a.lastRun.status].sortOrder -
+          JobStatus[b.lastRun.status].sortOrder;
+    }
+  }
+
+  jobSortFunction(prop, direction) {
```

Comment:
After me and @weblancaster discussed offline, we'll keep this naming as-is because in other parts of the codebase we follow this convention for functions we pass to `Array#sort()` (although probably a strange naming pattern) 👍